src/Controller/CaseController.php

```diff
@@ -160,14 +159,12 @@ public function summary(BoardRepository $boardRepository, CaseEntity $case, Note
         $communications = $digitalPostRepository->findByEntity($case, [], ['createdAt' => Criteria::DESC], $numberOfShownItems);
 
         $suitableBoards = $boardRepository->findDifferentSuitableBoards($case->getBoard());
+
         // Check if case is deletable
-        $isDeletable = true;
-        // Case is not deletable if case is on active agenda
-        foreach ($case->getAgendaCaseItems() as $agendaCaseItem) {
-            if (AgendaStatus::FINISHED !== $agendaCaseItem->getAgenda()->getStatus()) {
-                $isDeletable = false;
-            }
-        }
+        // If it is in hearing or has been in hearing it is not deletable
+        $hasBeenInHearing = $case->getHearing() && ($case->getHearing()->getStartedOn() || $case->getHearing()->getFinishedOn());
+        // If it has been on agenda or is on agenda it is also not deletable
+        $isDeletable = $case->getAgendaCaseItems()->isEmpty() && !$hasBeenInHearing;
```

Comment:
Remember to also check these conditions in the actual `delete` action. Otherwise a malicious user can just delete the case anyway.

Reply:
Good catch. Added check in `CaseController::delete()` method in 6710faf8c6b8a86d3a558bb2ac7394720c8cb85f.